class.jetpack.php

```diff
@@ -822,6 +835,32 @@ function update_jetpack_main_network_site_option() {
 	}
 
 	/**
+	 * Get back if the current site is single user site.
+	 *
+	 * @return bool
+	 */
+	public function is_single_user_site() {
+
+		$user_query = new WP_User_Query( array(
+			'blog_id' => get_current_blog_id(),
+			'fields'  => 'ID',
+			'number' => 2
+		) );
+		return ( (int) $user_query->get_total() > 1 ? true : false );
+	}
+
+	/**
+	 * Invalides the transient as well as triggers the update of the mock option.
+	 *
+	 * @return null
+	 */
+	function is_single_user_site_invalidate() {
+		do_action( 'update_option_jetpack_single_user_site', 'jetpack_single_user_site', (bool) $this->is_single_user_site() );
```

Comment:
Nice trick :)

This give me the idea that for constants we could have a hash of all the constants together, that we can check if it changes in order to fire a sync instead of doing it just on heartbeat.